### ISSUE TITLE
Task - "integrationTest" always assumes, it needs to start tomcat before and stop after tests

### ISSUE THREAD

**Issue by mkornipati** [saladinkzn/gretty#10]
This is same as created @  https://github.com/akhikhl/gretty/issues/244

---

I am referring to the ["Gretty Configuration"](http://akhikhl.github.io/gretty-doc/Gretty-configuration.html) and understood that the default value for `integrationTestTask` is `integrationTest`. 

We got few projects where we don't need the Tomcat needs to be running for integrationTest Task. and actually name of the integrationTest task  is `integrationTest`.  I am looking for Gretty Configurations to see,  if I can say, I don't need Tomcat start and stop for my "integrationTest" task. Can you point me to right configuration if any.

Here are few other observations. 
I tried setting the `integrationTestTask` as `null` or `''`, but I can still see the Tasks - `appBeforeIntegrationTest`and `appAfterIntegrationTest` getting executed. 
The reason seems to be because `appBeforeIntegrationTest.integrationTestTaskAssigned`, `appAfterIntegrationTest.integrationTestTaskAssigned`  is always`true` and it is not updated to 'false' when I am updating value of `integrationTestTask` to null or empty string?

**Comment by saladinkzn** [comment]
Hello

Do you set integration task to null in gretty configuration? Like:
```groovy
gretty {
    integrationTestTask = null
}
```
?

2016-04-14 2:31 GMT+03:00 Mohan Kornipati <notifications@github.com>:

**Comment by saladinkzn** [comment]
Hello.

Do you set integration task to null in `gretty` configuration? Like:

```
gretty {
     integrationTestTask = null
}
```

What I see here, that there's a bug if you don't set `integrationTestTask` to null in Gretty configuration you  actually get it used in AppBeforeIntegrationTestTask if none or falsy is assigned in it.

**Comment by mkornipati** [comment]
Yes @saladinkzn. As mentioned in the the description, I tried setting the integrationTestTask as `null`or `''` and both not worked.

**Comment by saladinkzn** [comment]
OK, I'll check it out today

2016-04-14 15:31 GMT+03:00 Mohan Kornipati notifications@github.com:

**Comment by mkornipati** [comment]
Thanks @saladinkzn  and just to let you know, currently, as a workaround we are removing the `appBeforeIntegrationTest` from `integrationTest.dependsOn`and `appAfterIntegrationTest` from `integrationTest.finalizedBy`

```
     project.integrationTest.dependsOn.remove(project.tasks['appBeforeIntegrationTest'])
     project.integrationTest.finalizedBy.values.remove(project.tasks['appAfterIntegrationTest'])
```

**Comment by saladinkzn** [comment]
Well, I've tested [gist](https://gist.github.com/saladinkzn/c1ab77c702f52d0e4b161152abe48d9a) and it's working as expected (server is not started when integrationTest task is called using `./gradlew integrationTest`

Could you provide a minimal example when appBeforeIntegrationTest is called even when integrationTestTask is set to null?
